src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -726,46 +795,80 @@ gl_sframe activity_classifier::perform_inference(data_iterator *data) const {
       get_inference_config(prediction_window),
       nn_spec_->export_params_view());
 
-  while (data->has_next_batch()) {
-    // Obtain the next batch of inputs.
-    data_iterator::batch inputs =
-        data->next_batch(read_state<flex_int>("batch_size"));
+  // To support double buffering, use a queue of pending inference results.
+  std::queue<result> pending_batches;
+  
+  auto pop_until_size = [&](size_t remaining) {
 
-    // Send the inputs to the model.
-    // TODO: Implement double buffering.
-    std::map<std::string, shared_float_array> results =
-        backend->predict({ { "input", inputs.features } });
+    while (pending_batches.size() > remaining) {
 
-    // Copy the (float) outputs to our (double) buffer and add to the SArray.
-    const shared_float_array& output = results.at("output");
-    for (size_t i = 0; i < inputs.batch_info.size(); ++i) {
-      shared_float_array output_chunk = output[i];
-      data_iterator::batch::chunk_info info = inputs.batch_info[i];
+      // Pop one batch from the queue.
+      result batch = pending_batches.front();
+      pending_batches.pop();
+      
+      for (size_t i = 0; i < batch.data_info.batch_info.size(); ++i) {
+      
+      shared_float_array output_chunk = batch.output_info[i];
+      data_iterator::batch::chunk_info info = batch.data_info.batch_info[i];
```

Comment:
The body of the for loop should be indented relative to the `for` statement itself